### PR TITLE
Bugfix/authorization fix

### PR DESCRIPTION
**Pull request by dapal-003**
* Fixed authorization not being set resulting in the user never getting logged in.
* Fixed web socket not being opened at the right time resulting in having the user connect twice for the first change.
* Anonymized email in logging.
* Fixed typo that resulted in users never being able to log out

### CODE REVIEW

File: WFInfo/Data.cs [hunk 1]
```diff
@@ -1319,10 +1317,8 @@ private void SendMessage(string data)
                 if (marketSocket.ReadyState == WebSocketState.Closed || marketSocket.ReadyState != WebSocketState.Open)
                 {
                     marketSocket.Close();
-                    OpenWebSocket();
+                    bool result = OpenWebSocket().Result;
```
Comment (dimon222) [inline WFInfo/Data.cs:1320]: Is there specific reason we don't check this "result" output?

File: WFInfo/Data.cs [hunk 2]
```diff
@@ -1159,7 +1157,7 @@ public void SetJWT(HttpResponseHeaders headers)
         {
             foreach (var item in headers)
             {
-                if (!item.Key.Contains("Authorization")) continue;
+                if (!item.Key.Contains("authorization")) continue;
```
Comment (dimon222) [inline WFInfo/Data.cs:1160]: Weird, is it intended?

File: WFInfo/Data.cs [hunk 3]
```diff
@@ -1103,7 +1101,7 @@ public async Task<bool> OpenWebSocket()
             Main.AddLog("Connecting to websocket");
             marketSocket.SslConfiguration.EnabledSslProtocols = SslProtocols.Tls12;
 
-            if (marketSocket.IsAlive || marketSocket.ReadyState == WebSocketState.Connecting)
+            if (marketSocket.IsAlive) //|| marketSocket.ReadyState == WebSocketState.Connecting
```
Comment (dimon222) [inline WFInfo/Data.cs:1104]: I suggest we remove the commented part altogether, no point to keep it there if its not needed anymore.

File: WFInfo/Data.cs (hunk 1 above)
Comment (dapal-003) [reply WFInfo/Data.cs:1320]: The result is not used to verify the status of the task, rather to make sure the asynchronous task has finished before moving on. The previous way of handling it caused some errors to throw in edge cases, this way of handling it solves that.

File: WFInfo/Data.cs [hunk 4]
```diff
@@ -1087,7 +1083,9 @@ public async Task GetUserLogin(string email, string password)
             }
             else
             {
-                throw new Exception("GetUserLogin, " + responseBody + $"Email: {email}, Pw length: {password.Length}");
+                Regex rgx = new Regex("[a-zA-Z0-9]");
+                string censoredEmail = rgx.Replace(email, "*");
+                throw new Exception("GetUserLogin, " + responseBody + $"Email length: {censoredEmail}, Pw length: {password.Length}");
```
Comment (dimon222) [inline WFInfo/Data.cs:1088]: This is not length tho, this is an email. Perhaps we want to change print `censoredEmail.length()`?

File: WFInfo/Data.cs (hunk 2 above)
Comment (dapal-003) [reply WFInfo/Data.cs:1160]: Not sure, don't know when it changed either

File: WFInfo/Data.cs (hunk 4 above)
Comment (dapal-003) [reply WFInfo/Data.cs:1088]: Was a previous change, forgot to remove the length again from the debug line

File: WFInfo/Data.cs (hunk 3 above)
Comment (dapal-003) [reply WFInfo/Data.cs:1104]: +1

**Pull request merged**